### DISCUSSION TITLE
Динамическая таблица

### DISCUSSION THREAD

**Discussion by fonderkin** [totumonline/totum-ru-issues-and-discussions#606] [Как это сделать?] [answered]
Здравствуйте! Делаю программу для учета арендного оборудования. Собрали такую конфигурацию, есть 2 таблицы: товары и заказы (дата начала, дата конец, состав заказа товары).  Хотим сделать третью таблицу которая по дате будет выдавать товары не участвующие в заказах (по факту в аренде) в этот день.

Какие были мысли по реализации

1. Сделать таблицу с полем дата
2. Делаем поле insert в котором пишем логику по условию
`найди заказы, промежуток которых попадает дата и делаем выборку по товарам кроме найденных позиций из заказа.`
Код получился, выдает id свободных товаров
3. Нужно их записать в таблицу, вот как раз с этим и не понятно, что нужно использовать чтобы на базе результатов кода отрисовывать таблицу?

Пытался решить через [Автозаполнение как в примере](https://www.youtube.com/watch?v=OUihzx5Ai_c&t=9s) (просто прокидывать из результатов поля) не работает.

**Comment by ttm-alexeypolunin** [comment]
insert (автозаполнение) работает в типе таблицы Временная и Расчетная. В вашем случае нужна Временная.

ну и если он там ошибку говорит, то какую?

**Reply by fonderkin** [reply]
Получилось, была ошибка в запросе. Еще не понятен момент, как отрисовать динамические столбцы, грубо говоря на итого нужна таблица

Первая колонка название товара
Далее динамиеские колонки с датой 13.10.2024 | 14.10.2024 | .... 

Как такое создать?

**Reply by LostAlly** [reply]
Это называется сводная таблица, вроде как тут есть ответ, сделайте поиск по фразе сводная таблица.

**Reply by ttm-alexeypolunin** [reply, marked as answer]
Динамическое поле есть в PRO: https://ru.docs.totum.online/dynamic-fields
